### PR TITLE
Add new flag to collect minimal resources

### PR DESCRIPTION
**Pull request by Sheetalpamecha**
Creating a flag to limit data collection to only major CRDs and CSVs, reducing time taken and log clutter.
It collects minimum required data and have kept the alignment similar to other flags for streamlined parsing.

### CODE REVIEW

**Comment by Sheetalpamecha** [comment]
@yati1998 and @black-dragon74 Please take a look.

File: collection-scripts/gather [hunk 1]
```diff
@@ -93,6 +99,7 @@ Options:
   -cl, --ceph-logs          Collect ceph daemon, kernel, journal logs and crash reports
   -ns, --namespaced         Collect namespaced resources
   -cs, --clusterscoped      Collect clusterscoped resources
+  --minimal-crds            Collect only relevant CRDs and CSVs
```
Comment (yati1998) [inline collection-scripts/gather:102]: how about adding short form for it too?

**Comment by Sheetalpamecha** [comment]
Hi Yati, True there will be redundancy, As we want to highlight few configurations only and we don't want the logs and other information collected. 
Here is the epic link - https://issues.redhat.com/browse/RHSTOR-5936

File: collection-scripts/gather (hunk 1 above)
Comment (Sheetalpamecha) [reply collection-scripts/gather:102]: Sure, How about -mr, --minimal-resources?

Comment (black-dragon74) [reply collection-scripts/gather:102]: MG always collects CRDs or resources. Let's omit it and call it `-m, minimal`?

File: collection-scripts/gather [hunk 2]
```diff
@@ -197,6 +204,12 @@ if [ "$clusterscoped" == true ] && [ "$odf" == false ]; then
     pids+=($!)
 fi
 
+if [ "$minimal_crds" == true ] && [ "$odf" == false ]; then
```
Comment (black-dragon74) [inline collection-scripts/gather:207]:
```suggestion
if [ "$minimal_crds" == true ]; then
```
We do not need to check for `odf == false` here as the minimal script is not part of the odf collection. The reason we check for `odf == false` is so we do not collect resources that are collected with `--odf` mode twice.

File: collection-scripts/gather (hunk 1 above)
Comment (Sheetalpamecha) [reply collection-scripts/gather:102]: Done

File: collection-scripts/gather (hunk 2 above)
Comment (black-dragon74) [reply collection-scripts/gather:207]: Umm.. I think you forgot to commit the changes..

P.S: We should not mark conversations as resolved by ourselves as it makes it harder to keep track of changes for reviewers. Ideally, the person who left the comment resolves it upon confirming the changes :)

Comment (Sheetalpamecha) [reply collection-scripts/gather]: Noted, will be careful with future PRs.
Pushed the correct changes

File: collection-scripts/gather [hunk 3]
```diff
@@ -93,6 +99,7 @@ Options:
   -cl, --ceph-logs          Collect ceph daemon, kernel, journal logs and crash reports
   -ns, --namespaced         Collect namespaced resources
   -cs, --clusterscoped      Collect clusterscoped resources
+  -m,  --minimal            Collect only relevant CRDs and CSVs
```
Comment (black-dragon74) [inline collection-scripts/gather:102]: A more specific description would be nice. "only relevant" sounds a bit vague?

File: collection-scripts/gather [hunk 4]
```diff
@@ -197,6 +204,12 @@ if [ "$clusterscoped" == true ] && [ "$odf" == false ]; then
     pids+=($!)
 fi
 
+if [ "$minimal" == true ]; then
+    echo "Collect minimal resource files..."
+    gather_minimal_resources ${BASE_COLLECTION_PATH} &
```
Comment (black-dragon74) [inline collection-scripts/gather:209]:
```suggestion
    gather_minimal_resources &
```

File: collection-scripts/gather (hunk 3 above)
Comment (Sheetalpamecha) [reply collection-scripts/gather]: Done

File: collection-scripts/gather (hunk 4 above)
Comment (Sheetalpamecha) [reply collection-scripts/gather]: Done

**Comment by yati1998** [comment]
@Sheetalpamecha I hope you have tested this change and verified that you are able to collect all the required crds using this flag?

**Comment by Sheetalpamecha** [comment]
Hi @yati1998 , Yes I have tested and this option is able to collect required crds.

File: collection-scripts/gather (hunk 4 above)
Comment (black-dragon74) [reply collection-scripts/gather:209]: It’s fine I guess. But if you look at actual scripts, none of them read this arg. They read the exported BASE_COLLECTION_PATH. 

Anyways, let’s keep it.

**Pull request merged**